src/schedule/ll_schedule.c

```diff
@@ -353,6 +354,33 @@ static int schedule_ll_task(void *data, struct task *task, uint64_t start,
 
 	pdata->period = period;
 
+	/* for full synchronous domain, calculate ratio and initialize skip_cnt for task */
+	if (sch->domain->full_sync) {
+		pdata->ratio = 1;
+		pdata->skip_cnt = (uint16_t)(-1);
+
+		/* get the registrable task */
+		list_for_item(tlist, &sch->tasks) {
+			curr_task = container_of(tlist, struct task, list);
+			pipe_task = pipeline_task_get(curr_task);
+
+			/* registrable task found */
+			if (pipe_task->registrable) {
+				registrable_task = curr_task;
+				break;
+			}
+		}
+
+		/* we found a registrable task */
+		if (registrable_task) {
+			reg_pdata = ll_sch_get_pdata(registrable_task);
+
+			/* the assumption is that the registrable task has the smallest period */
+			assert(period >= reg_pdata->period);
+			pdata->ratio = period / reg_pdata->period;
```

Comment:
what will happen if the ratio is decimal, e.g. registrable_task with period 3ms and the deep buffer one with period 5ms? Will we still hit similar issue with the PR applied?

Reply:
Based on how the dma_multi_chan_domain is written the assumption is that the other tasks' period is a multiple of the registrable task period. 
Also, the topology files confirm this (at least for imx).

Back to your original question: for 3 and 5 it could happen if the registrable task finishes the DMA transfer earlier than 15ms. But, it will not be fatal, since it can be scheduled latter (at 18ms).

This PR doesn't apply to this scenario, but to the one I described in the beginning.